modules/electrophysiology_browser/ajax/BIDSFile.class.inc

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Contains the BIDSFile class used for electrophysiological browser
+ *
+ * PHP Version 7
+ *
+ * @category EEG
+ * @package  EEG
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+/**
+ * This object represents a BIDS file in the physiological_file table.
+ *
+ * @category EEG
+ * @package  EEG
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class BIDSFile
+{
+    var $fileData   = array();
+    var $parameters = array();
+
+    /**
+     * Construct a BIDSFile
+     *
+     * @param integer $physiologicalFileID The PhysiologicalFileID to be loaded
+     */
+    function __construct($physiologicalFileID)
+    {
+        $db     =& Database::singleton();
```

Comment:
```suggestion
        $db     = \NDB_Factory::singleton()->database();
```
As per @AlexandraLivadas's [presentation](https://alexandralivadas.github.io/assets/GSOC_Presentation.pdf) (page 9) we should be using the Factory instead of the singleton directly. This allows for much easier creation of unit tests.

Reply:
Good callback, @johnsaigle 🎩